lib/ibmdb.js

```diff
@@ -409,32 +425,37 @@ IBMDB.prototype.updateOrCreate = IBMDB.prototype.save =
       // Execution for updateOrCreate requires running two
       // separate SQL statements.  The second depends on the
       // result of the first.
+      var where = {};
+      where[id] = data[id];
+
       var countStmt = new ParameterizedSQL('SELECT COUNT(*) AS CNT FROM ');
       countStmt.merge(tableName);
-      countStmt.merge(self.buildWhere(data));
+      countStmt.merge(self.buildWhere(model, where));
+      countStmt.noResults = false;
 
-      connection.query(countStmt.sql, countStmt.params,
-        function(err, countData) {
+      connection.query(countStmt, function(err, countData) {
+        if (err) {
+          debug('IBMDB.prototype.updateOrCreate (ERROR):%j', err);
+          return cb(err);
+        }
+
+        if (countData[0]['CNT'] > 0) {
+          stmt = self.buildUpdate(model, where, data);
+        } else {
+          stmt = self.buildInsert(model, data);
+        }
+
+        stmt.noResults = true;
+
+        connection.query(stmt, function(err, sData) {
           if (err) {
+            debug('IBMDB.prototype.updateOrCreate (ERROR):%j', err);
```

Comment:
same as this one, put it outside the `if (err)` on 451 and `if (err) return cb(err);`